src/thor/timedistancebssmatrix.cc

```diff
@@ -501,8 +487,8 @@ bool TimeDistanceBSSMatrix::UpdateDestinations(
 
     // Erase this edge from further consideration. Mark this destination as
     // settled if all edges have been found
-    dest.dest_edges.erase(dest_edge);
-    if (dest.dest_edges.empty()) {
+    dest.dest_edges_available.erase(pred.edgeid());
```

Comment:
the actual change is adding this `dest_edges_available` unordered_set and splitting it from the `percent_along` info. the former we'll reset on every origin, the latter will stay for the entire request.